### PR TITLE
Update 'Chef Server' to 'Chef Infra Server' in the docs

### PR DESCRIPTION
**Pull request by btm**
All instances of 'Chef Server' are updated to 'Chef Infra Server' except where they refer to UI elements in the software which wouldn't have been changed.
    
This is to align with our use of 'Chef Infra' to describe the 'Chef' project to provide more specific clarity to users.
    
A few but not all 'Chef Client' to 'Chef Infra Client' changes are also made in this commit.

### CODE REVIEW

File: components/automate-chef-io/content/docs/client-runs.md [hunk 1]
```diff
@@ -1,235 +1,239 @@
-+++
```
Comment (susanev) [inline components/automate-chef-io/content/docs/client-runs.md:1]: what changed in this file? im so confused.

File: components/automate-chef-io/content/docs/iam-v2-guide.md [hunk 2]
```diff
@@ -371,7 +371,7 @@ To create a project that contains all Effortless Infra nodes, create a ingest ru
 
 ![](/images/docs/effortless-project-rule.png)
 
-The above rule matches on a node's Chef Server field, which is set to `localhost`. This rule works because all Effortless Infra nodes list the `Chef Server` attribute as `localhost`.
+The above rule matches on a node's Chef Infra Server field, which is set to `localhost`. This rule works because all Effortless Infra nodes list the `Chef Server` attribute as `localhost`.
```
Comment (susanev) [inline components/automate-chef-io/content/docs/iam-v2-guide.md:374]: this can be updated to `Chef Infra Server` as well, as #3352 will fix it in the ui

**Comment by susanev** [comment]
we need to fix it in these places too
https://automate.chef.io/docs/iam-v2-overview/#project-ingest-rule-conditions
and a bunch more in the guide https://automate.chef.io/docs/iam-v2-guide/
and the guide prolly needs a new screenshot

i can do these in a follow up pr if yall want

File: components/automate-chef-io/content/docs/client-runs.md (hunk 1 above)
Comment (kagarmoe) [reply components/automate-chef-io/content/docs/client-runs.md:1]: I'm also not sure, but it is 4 lines longer.

**Comment by kagarmoe** [comment]
@susanev Thanks! If you have the work in your other PR, then rebase & we'll move on. Otherwise, make a card with the docs changes you need & hand it over to @mjingle.

**Comment by susanev** [comment]
alright #3352 has the leftover doc updates for chef infra server

**Pull request merged**